eth/client.go

```diff
@@ -150,306 +226,601 @@ func NewTransactOptsForAccount(account accounts.Account, passphrase string, keyS
-		return nil, err
+		glog.Errorf("Error checking if current round initialized: %v", err)
+		return false, err
 	}
 
-	glog.Infof("[%v] Submitted transaction %v. Bond %v LPTU to %v at CR %v CN %v CRSB %v CB %v", c.account.Address.Hex(), tx.Hash().Hex(), amount, toAddr.Hex(), cr, cn, crsb, cb)
-	return tx, nil
+	return initialized, nil
 }
 
-func (c *Client) ValidRewardTimeWindow() (bool, error) {
-	return c.protocolSession.ValidRewardTimeWindow(c.account.Address)
+func (c *Client) IsActiveTranscoder() (bool, error) {
+	return c.bondingManagerSession.IsActiveTranscoder(c.account.Address)
 }
 
-func (c *Client) Reward() (*types.Transaction, error) {
-	tx, err := c.protocolSession.Reward()
-
+func (c *Client) TranscoderBond() (*big.Int, error) {
+	transcoder, err := c.bondingManagerSession.Transcoders(c.account.Address)
 	if err != nil {
-		glog.Errorf("Error calling reward: %v", err)
 		return nil, err
 	}
 
-	cr, cn, crsb, cb, err := c.RoundInfo()
+	return transcoder.BondedAmount, nil
+}
 
-	if err != nil {
-		glog.Errorf("Error getting round info: %v", err)
-		return nil, err
-	}
+func (c *Client) TranscoderStake() (*big.Int, error) {
+	return c.bondingManagerSession.TranscoderTotalStake(c.account.Address)
+}
 
-	glog.Infof("[%v] Submitted transaction %v. Reward at CR %v CN %v CRSB %v CB %v", c.account.Address.Hex(), tx.Hash().Hex(), cr, cn, crsb, cb)
-	return tx, nil
+func (c *Client) TranscoderStatus() (uint8, error) {
+	return c.bondingManagerSession.TranscoderStatus(c.account.Address)
 }
 
-func (c *Client) Job(streamId string, transcodingOptions [32]byte, maxPricePerSegment *big.Int) (*types.Transaction, error) {
-	tx, err := c.protocolSession.Job(streamId, transcodingOptions, maxPricePerSegment)
+func (c *Client) DelegatorStake() (*big.Int, error) {
+	return c.bondingManagerSession.DelegatorStake(c.account.Address)
+}
+
+func (c *Client) LastRewardRound() (*big.Int, error) {
+	transcoderDetails, err := c.bondingManagerSession.Transcoders(c.account.Address)
 
 	if err != nil {
-		glog.Errorf("Error creating job: %v", err)
+		glog.Errorf("Error getting transcoder details: %v", err)
 		return nil, err
 	}
 
-	glog.Infof("[%v] Submitted transaction %v. Creating a job for stream id %v", c.account.Address.Hex(), tx.Hash().Hex(), streamId)
-	return tx, nil
-}
-
-func (c *Client) JobDetails(id *big.Int) (*big.Int, [32]byte, *big.Int, common.Address, common.Address, *big.Int, error) {
-	return c.protocolSession.GetJob(id)
+	return transcoderDetails.LastRewardRound, nil
 }
 
-func (c *Client) EndJob(id *big.Int) (*types.Transaction, error) {
-	return c.protocolSession.EndJob(id)
-}
-
-func (c *Client) ClaimWork(jobId *big.Int, startSegmentSequenceNumber *big.Int, endSegmentSequenceNumber *big.Int, transcodeClaimsRoot [32]byte) (*types.Transaction, error) {
-	tx, err := c.protocolSession.ClaimWork(jobId, startSegmentSequenceNumber, endSegmentSequenceNumber, transcodeClaimsRoot)
+func (c *Client) ProtocolTimeParams() (*struct {
```

Comment:
Again, totally open question due to less familiarity with convention here...but better to return an anonymous struct or a typed one for reference and documentation purposes?